### PR TITLE
Add navigator property to PullDownButton

### PR DESCRIPTION
**Pull request by gborges9**
This PR introduces a new navigator property to the package, allowing users to specify a custom navigator when pushing a route. This enhancement is particularly useful for scenarios involving nested navigation.

### CODE REVIEW

**Comment by notDmDrl** [comment]
Hi @gborges9, 

Before I look over this, I want to know why something like this will be useful since I have not seen anything like this for any menus or dialogs in the Flutter repo. My knowledge of custom navigators is non-existent 🥲.

Could you, by any chance, provide some screenshots or videos showing how this PR solves the problems you described in the PR message?

Thanks

**Comment by gborges9** [comment]
Hey @notDmDrl,

When using a nested navigator, it may not occupy the entire screen. This causes the popup menu to remain visible even if you click outside the navigator where the popup menu route was pushed.

Here's an example to illustrate the issue:
https://github.com/notDmDrl/pull_down_button/assets/37756671/18994ef7-08c5-4be3-aef2-5fd890dd1a4a

One straightforward solution to this edge case is to provide the root navigator manually:
https://github.com/notDmDrl/pull_down_button/assets/37756671/57adda62-44cf-48c1-b8cd-cee790e3e85e


Alternatively, you could render the menu as an overlay entry and use a custom widget to wrap the app and handle the menu overlay entries. However, this approach would require some refactoring

File: lib/src/pull_down_button.dart [hunk 1]
```diff
@@ -509,7 +519,7 @@ Future<VoidCallback?> _showMenu<VoidCallback>({
   required double menuOffset,
   required ScrollController? scrollController,
 }) {
-  final navigator = Navigator.of(context);
+  navigator ??= Navigator.of(context);
```
Comment (notDmDrl) [inline lib/src/pull_down_button.dart:522]: I don't like this reassignment here. Rather create a new local variable like this `final resolvedNavigator = navigator ?? Navigator.of(context);`

File: lib/src/pull_down_button.dart [hunk 2]
```diff
@@ -336,6 +337,11 @@ class PullDownButton extends StatefulWidget {
   @experimental
   final Alignment? animationAlignmentOverride;
 
+  /// Custom navigator used to show the pull-down menu.
```
Comment (notDmDrl) [inline lib/src/pull_down_button.dart:340]: Could you also add what purpose this property has? Maybe mention that nested navigator use case?

**Comment by notDmDrl** [comment]
> is to provide the root navigator manually

Wouldn't it be better to just expose the `rootNavigator` property of `Navigator.of(context)` in `_showMenu` as it defaults to `false`?

File: lib/src/pull_down_button.dart (hunk 1 above)
Comment (notDmDrl) [reply lib/src/pull_down_button.dart:522]: Also see https://github.com/notDmDrl/pull_down_button/pull/49#issuecomment-2161438989

**Review by notDmDrl — CHANGES_REQUESTED** [review]
Overall looks good, only minor stuff code-wise.

For overall implementation please see https://github.com/notDmDrl/pull_down_button/pull/49#issuecomment-2161438989, maybe it will be a better solution

**Comment by notDmDrl** [comment]
Hey @gborges9 

I've noticed that you decided to go with `rootNavigator` idea. I assume this solves the issue this PR tries to fix? 

If it does and this is ready to be merged please let me know, I will try to merge this ASAP

**Comment by andychucs** [comment]
by the way, this PR also fixed a vertical offset problem in nested navigator.
截屏2024-08-03 09 49 26
截屏2024-08-03 09 49 48

**Comment by notDmDrl** [comment]
Hi @gborges9 

Sorry for the late response! I will close this PR as I implemented the `useRootNavigator` feature myself while working on update **0.10.0** (now live). You are, however, still credited for the change in the changelog 🙇‍♂️

Thanks again 😃

**Pull request closed without merging**